tests/integration/test_pip.py

```diff
@@ -1,15 +1,48 @@
 import logging
 import os
+import subprocess
+import time
 from pathlib import Path
-from typing import List
+from typing import Iterator, List
 
 import pytest
+import requests
 
 from . import utils
 
 log = logging.getLogger(__name__)
 
 
+TEST_LOCAL_PYPISERVER = os.getenv("CACHI2_TEST_LOCAL_PYPISERVER") == "true"
+
+
+@pytest.fixture(scope="module")
+def local_pypiserver() -> Iterator[None]:
+    if not TEST_LOCAL_PYPISERVER:
+        yield
+        return
+
+    pypiserver_dir = Path(__file__).parent.parent / "pypiserver"
+    proc = subprocess.Popen([pypiserver_dir / "start.sh"])
+
+    pypiserver_port = os.getenv("PYPISERVER_PORT", "8080")
+    for _ in range(60):
+        time.sleep(1)
+        try:
+            resp = requests.get(f"http://localhost:{pypiserver_port}")
+            resp.raise_for_status()
+            log.debug(resp.text)
+            break
+        except requests.RequestException as e:
+            log.debug(e)
+    else:
+        proc.terminate()
+        raise RuntimeError("pypiserver didn't start fast enough")
+
+    yield
+    proc.terminate()
+
+
```

Comment:
nitpick: not a big deal but strictly from correctness POV we should clean up the zombie process explicitly despite either Python runtime or whatever the parent is will take care of it.

Reply:
fixed